src/main/java/nextstep/ladder/domain/Ladder.java

```diff
@@ -3,18 +3,33 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.stream.Collectors;
-import java.util.stream.IntStream;
+import java.util.stream.Stream;
 
-public class Ladder {
+import static nextstep.ladder.domain.Direction.STAY;
 
+public class Ladder {
     private final List<Line> lines;
 
     public Ladder(Height height, int countOfPerson) {
-        this.lines = IntStream.range(0, height.height())
-                .mapToObj(i -> LineFactory.createLine(countOfPerson))
+        this.lines = Stream.generate(() -> createLine(countOfPerson))
+                .limit(height.height())
                 .collect(Collectors.toList());
     }
 
+    private static Line createLine(int countOfPerson) {
+        if(countOfPerson == 1){
+            return new Line(STAY);
+        }
+        return new Line(countOfPerson);
+    }
+
+    public int climb(int position) {
```

Comment:
```suggestion
    public int climb(final int position) {
```

참조변수를 수정하는것에 대해서 어떻게 생각하시나요?

조금은 번외의 질문으로, 불변객체를 사용하면 어떠한 장점이 있을까요?

Reply:
참조변수를 수정할 수 있다면 

제가 아는 불변객체 장점은 아래와 같습니다!!
- 객체 생성 이후 객체의 상태를 변경할 수 없어 Thread safe
- 값 수정을 할 수 없어 사이드 이펙트 감소
  - A 개발자가 만든 값을 B 개발자가 수정하면 디버깅 시간 소요


```java
    public Direction climb(int position) {
        Direction direction = Direction.of(position);
        for (Line line : lines) {
            direction = direction.plus(line.move(position));
        }
        return direction;
    }```
climb 메소드에서 1차 피드백때 위와 같이 피드백을 주셨는데요..!! 몇시간을 생각해봐도 의도하신 로직이 잘 이해가 안갑니다 ㅠ.ㅠ
- 인자로 건네주는 position 값은 참여자의 시작 위치(4명이면 0,1,2,3), but 코드상 `Direction.of(positon)` 할시 2,3일경우 exception 발생
- Directon 값은 -1,0,1 만 존재하는데 climb 메소드의 반환 값으로 Direction을 반환(?)

그래서 3차 수정때는 아래와 같이 수정하였습니다. 불변은 아니지만 계속해서 +=만 하지 않고 line에 메시지를 보내서 사다리를 타면서 라인별 결과값을 받도록 수정 하였습니다..

```java
    public int climb(int position) {
        for (Line line : lines) {
            position = line.move(position);
        }
        return position;
    }
```

※ 자세한 내용은 3차 리뷰 내용 중 commit 내용 확인 부탁 드립니다..!